### PR TITLE
feat: Add model version

### PR DESCRIPTION
**Pull request by etredal**

### CODE REVIEW

**Review by ishaansehgal99 — APPROVED** [review]

File: docker/presets/llama-2/Dockerfile [hunk 1]
```diff
@@ -26,8 +26,10 @@ RUN pip install 'uvicorn[standard]'
 ARG WEIGHTS_PATH
 ARG MODEL_TYPE
 ARG VERSION
-# Write the version to a file
-RUN echo $VERSION > /workspace/llama/version.txt
+ARG IMAGE_NAME
+
+# Write metadata to model_info.txt file
+RUN echo -e "$VERSION\n$IMAGE_NAME" > /workspace/llama/model_info.txt
```
Comment (ishaansehgal99) [inline docker/presets/llama-2/Dockerfile:32]: small nit - can the format of this be
$IMAGE_NAME:$VERSION

Comment (etredal) [reply docker/presets/llama-2/Dockerfile]: It is fixed

**Comment by ishaansehgal99** [comment]
There are two things left to do: 
1. CI/CD: Bump up all the model versions by 0.0.1 in supported_models.yaml, this will trigger the build pipeline
2. Test: Add test for the new endpoint in `e2e-preset-test.yaml` and `text-generation/tests/test_inference_api.py` (would also recommend testing the endpoint in own env)

**Comment by etredal** [comment]
@ishaansehgal99

**Comment by etredal** [comment]
@ishaansehgal99  Im also thinking the model_info.txt may be better off as a JSON file if it's not really needing to be human readable.  Besides that I have pushed new changes and was able to use the existing MODEL_VERSION for the URL we need.

File: docker/presets/llama-2/Dockerfile [hunk 2]
```diff
@@ -26,8 +28,11 @@ RUN pip install 'uvicorn[standard]'
 ARG WEIGHTS_PATH
 ARG MODEL_TYPE
 ARG VERSION
-# Write the version to a file
-RUN echo $VERSION > /workspace/llama/version.txt
+ARG IMAGE_NAME
+ARG MODEL_VERSION
+
+# Write metadata to model_info.txt file
+RUN echo -e "Model Type: $MODEL_TYPE\nVersion: $VERSION\nImage Name: $IMAGE_NAME\nModel Version URL: $MODEL_VERSION" > /workspace/llama/model_info.txt
```
Comment (ishaansehgal99) [inline docker/presets/llama-2/Dockerfile:35]: One more field here called REVISION_ID which is the hash at the end of the MODEL_VERSION

File: docker/presets/tfs-onnx/Dockerfile [hunk 3]
```diff
@@ -4,12 +4,14 @@ FROM mcr.microsoft.com/aifx/acpt/stable-ubuntu2004-cu118-py38-torch211
...
+ARG IMAGE_NAME
+ARG MODEL_VERSION
 
 # Set the working directory
 WORKDIR /workspace/tfs
 
-# Write the version to a file
-RUN echo $VERSION > /workspace/tfs/version.txt
+# Write metadata to model_info.txt file
+RUN echo -e "Model Type: $MODEL_TYPE\nVersion: $VERSION\nImage Name: $IMAGE_NAME\nModel Version URL: $MODEL_VERSION" > /workspace/tfs/model_info.txt
```
Comment (ishaansehgal99) [inline docker/presets/tfs-onnx/Dockerfile:14]: Same as above

File: docker/presets/llama-2/Dockerfile (hunk 2 above)
Comment (ishaansehgal99) [inline docker/presets/llama-2/Dockerfile:35]: Change to version.json instead of txt

File: presets/inference/llama2-chat/inference_api.py [hunk 4]
```diff
@@ -191,6 +194,13 @@ def get_metrics():
         except Exception as e:
             return {"error": str(e)}
 
+    @app_main.get("/version")
+    def get_version():
+        with open(f"/workspace/llama/{MODEL_INFO}", "r") as f:
+            model_name = f.read()
+
+        return {"version": model_name}
```
Comment (ishaansehgal99) [inline presets/inference/llama2-chat/inference_api.py:202]: read and return json now

File: presets/models/supported_models.yaml [hunk 5]
```diff
@@ -3,28 +3,29 @@ models:
   - name: llama-2-7b
     type: llama2-completion
     runtime: llama-2
-    tag: 0.0.3
+    tag: 0.0.4
```
Comment (ishaansehgal99) [inline presets/models/supported_models.yaml:6]: update tags

File: docker/presets/llama-2/Dockerfile [hunk 6]
```diff
@@ -26,8 +28,12 @@ RUN pip install 'uvicorn[standard]'
...
 ARG MODEL_TYPE
 ARG VERSION
-# Write the version to a file
-RUN echo $VERSION > /workspace/llama/version.txt
+ARG IMAGE_NAME
+ARG MODEL_VERSION
+
+# Write metadata to model_info.json file
+RUN MODEL_VERSION_HASH=$(echo "$MODEL_VERSION" | awk -F'/' '{print $NF}') && \
+    echo "{\"Model Type\": \"$MODEL_TYPE\", \"Version\": \"$VERSION\", \"Image Name\": \"$IMAGE_NAME\", \"Model Version URL\": \"$MODEL_VERSION\", \"REVISION_ID\": \"$MODEL_VERSION_HASH\"}" > /workspace/llama/model_info.json
```
Comment (ishaansehgal99) [inline docker/presets/llama-2/Dockerfile:36]: Eliminating need to rely on awk in the image - 
does this work: 
RUN `MODEL_VERSION_HASH="${MODEL_VERSION##*/}"` && \
    echo "{\"Model Type\": \"$MODEL_TYPE\", \"Version\": \"$VERSION\", \"Image Name\": \"$IMAGE_NAME\", \"Model Version URL\": \"$MODEL_VERSION\", \"REVISION_ID\": \"$MODEL_VERSION_HASH\"}" > /workspace/llama/model_info.json

Comment (ishaansehgal99) [reply docker/presets/llama-2/Dockerfile:36]: test briefly in your shell

Comment (etredal) [reply docker/presets/llama-2/Dockerfile:36]: This works

**Review by ishaansehgal99 — APPROVED** [review]

**Comment by ishaansehgal99** [comment]
Included in #341 (To run pipelines)

**Pull request closed without merging**